### PR TITLE
Specify the build system in pyproject.toml

### PR DESCRIPTION
**Pull request by Jackenmen**
## What do these changes do?

Adds `build-backend` to `pyproject.toml` so that there's no warnings from pip when building from source (+ so that modern setuptools backend is used).

## Are there changes in behavior for the user?

pip will no longer show warning:
```
  WARNING: Missing build requirements in pyproject.toml for multidict==6.0.4 from https://files.pythonhosted.org/packages/4a/15/bd620f7a6eb9aa5112c4ef93e7031bcd071e0611763d8e17706ef8ba65e0/multidict-6.0.4.tar.gz.
  WARNING: The project does not specify a build backend, and pip cannot fall back to setuptools without 'wheel'.
```

## Related issue number

None

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Comment by Jackenmen** [comment]
I'm not sure if a change like this requires a history fragment, let me know :)

**Comment by webknjaz** [comment]
I think a misc change fragment would be a good idea. It would be weird to make a release without a changelog entry.

**Review by Jamim — APPROVED** [review]

File: CHANGES/802.misc.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Switched from setuptools's legacy backend (``setuptools.build_meta:__legacy__``)
```
Comment (webknjaz) [inline CHANGES/802.misc.rst:1]:
```suggestion
Switched from ``setuptools``' legacy backend (``setuptools.build_meta:__legacy__``)
```

File: CHANGES/802.misc.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+to the modern one (``setuptools.build_meta``) by actually specifying the
+the ``build-system.build-backend`` option in ``pyproject.toml``.
```
Comment (webknjaz) [inline CHANGES/802.misc.rst:3]:
```suggestion
the ``[build-system].build-backend`` option in :file:`pyproject.toml`.
```

File: CHANGES/802.misc.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Switched from ``setuptools``' legacy backend (``setuptools.build_meta:__legacy__``)
+to the modern one (``setuptools.build_meta``) by actually specifying the
+the ``[build-system].build-backend`` option in :file:`pyproject.toml`.
```
Comment (webknjaz) [inline CHANGES/802.misc.rst:3]:
```suggestion
the ``[build-system].build-backend`` option in :file:`pyproject.toml`
-- by :user:`Jackenmen`.
```

**Pull request merged**